File: app/src/main/java/za/co/kernelpanic/cloudy/di/submodules/FragmentBuildersModule.java
package za.co.kernelpanic.cloudy.di.submodules;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import za.co.kernelpanic.cloudy.ui.fragments.Forecast.WeatherFragment;

@Module
public  abstract class FragmentBuildersModule {

    /*
     * Our activity will be able to inject this using dagger
     */
    @ContributesAndroidInjector
    abstract WeatherFragment weatherFragment();
}

File: app/src/main/java/za/co/kernelpanic/cloudy/di/submodules/FragmentBuildersModule.kt
package za.co.kernelpanic.cloudy.di.submodules

import dagger.Module
import dagger.android.ContributesAndroidInjector
import za.co.kernelpanic.cloudy.ui.fragments.forecast.WeatherFragment

@Module
abstract class FragmentBuildersModule {

    /*
     * Our activity will be able to inject this using dagger
     */
    @ContributesAndroidInjector
    abstract fun weatherFragment(): WeatherFragment
}